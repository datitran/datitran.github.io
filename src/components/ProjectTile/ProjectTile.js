import React, { Component } from 'react';

import GitHubStats from '../GitHubStats/GitHubStats';
import GitHubService from '../../services/GitHubService';

import './ProjectTile.scss';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export default class ProjectTile extends Component {
    constructor(props) {
        super(props);

        this.state = {
            stats: {}
        };
    }

    componentWillMount() {
        GitHubService.getStats(this.props.projectData.ownerId, this.props.projectData.projectId).then((stats) => {
            let state = this.state;
            state.stats = stats;
            this.setState(state);
        }).catch((err) => {
            console.log('failed', err);
        })
    }

    openVideo() {
        console.log('Track: openVideo');
        window.open(this.props.projectData.videoUrl, '_blank');
    }

    goToRepo() {
        console.log('Track: goToRepo');
    }

    render() {
        return (
            <div className='projectTile__container'>
                <h3 className='projectTile__heading'>
                    <a className='projectTile__title' href={this.state.stats.html_url} target='_blank' rel='noreferrer noopener'>{this.props.projectData.title}</a>
                        <img className='projectTile__logo' src={require('../../static/logo_' + this.props.projectData.ownerLogo + '.png')} alt={this.props.projectData.ownerLogo}/>
                </h3>
                <p className='projectTile__description'>{this.props.projectData.description}</p>
                <button className='projectTile__videoButton' style={{display: this.props.projectData.videoUrl ? 'block' : 'none' }} onClick={this.openVideo.bind(this)}><FontAwesomeIcon name='play-circle' style={{ fontSize: '20px' }} /> Video Presentation</button>
                <div className='projectTile__content'>
                    <GitHubStats stats={this.state.stats}  language={this.state.stats.language}/>
                </div>
            </div>
        );
    }
}
