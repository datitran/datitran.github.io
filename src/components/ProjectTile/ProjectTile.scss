@import '../../styles/variables.scss';

.projectTile {
    &__container {
        display: flex;
        flex-direction: column;
        min-width: 300px;
        width: 30%;
        padding: 20px;
        border: 1px solid #ddd;
        margin: 0 1.5% 30px;
        box-sizing: border-box;

        transition: .2s;

        .gitHubStats__container {
            transition: background-color .2s;
        }

        &:hover {
            background-color: #f7f7f7;
            box-shadow: 0px 0px 8px rgba(0,0,0,0.2);
            .gitHubStats__container {
                background-color: #fff;
            }
        }

        @media screen and (max-width: 750px) {
            width: 100%;
        }

        @media screen and (max-width : 500px) {
            background-color: #f7f7f7;

            .gitHubStats__container {
                background-color: #fff;
            }
        }
    }

    &__heading {
        margin: 0;
        text-transform: uppercase;
        font-size: 16px;
        padding-bottom: 5px;
        border-bottom: 2px solid #fff;
        margin-bottom: 5px;
        display: flex;
        align-items: center;
        font-weight: 700;       
    }

    &__title {
        flex: 1;        
    }

    &__logo {
        height: 30px;
    }

    &__description {
        font-size: 14px;
        line-height: 1.4;
        color: $darkgray;
    }

    &__videoButton {
        border: 0;
        background-color: transparent;
        padding:0;
        margin: 10px 0;
        font-style: italic;
        font-weight: 700;
        font-size: 14px;
        color: $darkgray;
        cursor: pointer;
    }

    &__github {
        background-color: #fff;
    }

    &__content {
        display: flex;
        flex-direction: row;
        flex: 1;
        align-items: flex-end;
        margin-top: 5px;
    }
}