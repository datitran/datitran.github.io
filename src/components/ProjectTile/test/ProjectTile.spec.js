import React from 'react';
import renderer from 'react-test-renderer';
import ProjectTile from '../ProjectTile';

describe('ProjectTile component', () => {
    it('should render as expected', () => {
        const renderedComponent = renderer.create(
            <ProjectTile projectData={{}} />
        );
        expect(renderedComponent.toJSON()).toMatchSnapshot();
    })
});