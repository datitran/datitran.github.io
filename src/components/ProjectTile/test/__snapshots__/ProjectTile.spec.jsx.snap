// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTile component should render as expected 1`] = `
<div
  className="projectTile__container"
>
  <h3
    className="projectTile__heading"
  >
    <a
      className="projectTile__title"
      href={undefined}
    />
    <a
      href={undefined}
      target="_blank"
    >
      <img
        className="projectTile__logo"
        src="/static/logo_undefined.png"
      />
    </a>
  </h3>
  <p
    className="projectTile__description"
  />
  <button
    className="projectTile__videoButton"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <span
      aria-hidden={true}
      className="fa fa-play-circle"
      style={
        Object {
          "fontSize": "20px",
        }
      }
    />
     Video Presentation
  </button>
  <div
    className="projectTile__content"
  >
    <div
      className="gitHubStats__container"
      onClick={[Function]}
    >
      <div
        className="gitHubStats__stat"
      >
        <p
          className="gitHubStats__icon"
        >
          <span
            aria-hidden={true}
            className="fa fa-code"
          />
        </p>
        <p
          className="gitHubStats__description"
        />
      </div>
      <div
        className="gitHubStats__stat"
      >
        <p
          className="gitHubStats__icon"
        >
          <span
            aria-hidden={true}
            className="fa fa-code-fork"
          />
        </p>
        <p
          className="gitHubStats__description"
        />
      </div>
      <div
        className="gitHubStats__stat"
      >
        <p
          className="gitHubStats__icon"
        >
          <span
            aria-hidden={true}
            className="fa fa-star"
          />
        </p>
        <p
          className="gitHubStats__description"
        />
      </div>
      <div
        className="gitHubStats__stat"
      >
        <p
          className="gitHubStats__icon"
        >
          <span
            aria-hidden={true}
            className="fa fa-users"
          />
        </p>
        <p
          className="gitHubStats__description"
        />
      </div>
    </div>
  </div>
</div>
`;
